conda_build/create_test.py

```diff
@@ -87,6 +88,28 @@ def create_shell_files(m):
                     f.write("if errorlevel 1 exit 1\n")
                 has_tests = True
 
+    libs = ensure_list(m.get_value('test/check_shared_libs_load', []))
+    if libs:
+        if sys.platform.startswith('win'):
+            lib_path = os.path.join(m.config.test_prefix, 'Library', 'bin')
+        else:
+            lib_path = os.path.join(m.config.test_prefix, 'lib')
+
+        with open(join(m.config.test_dir, name), 'a') as f:
+            f.write('\n\n')
+            for lib in libs:
+                lib_loc = os.path.join(lib_path, lib + get_shlib_ext())
+                # test existence
+                if sys.platform.startswith('win'):
+                    f.write('if not exist ' + lib_loc + ' exit 1\n')
+                else:
+                    f.write('test -f ' + lib_loc + '\n')
+                # test loading
+                f.write('python -c "import ctypes; ctypes.cdll[\'' + lib_loc + '\']"\n')
```

Comment:
I am very concerned that the entire Anaconda Distribution and conda-forge stacks be bootstrappable. This naturally leads to cyclic dependencies not being allowed. The C libraries and tools necessary to bootstrap `Python` (on Linux) are:
`m4 perl autoconf automake pkg-config zlib xz bzip2 tk libffi openssl ncurses readline sqlite yaml`

Making this new feature depend on Python precludes using it in any of those libraries. Python is being used to implement this because it is quick and easy and not because it is the correct way to do it.

We should compile simple test helper executables that takes as arguments e.g.: `libname1,symbol1,symbol2` `libname2,symbol3,symbol4`